packages/plugins/xray/src/index.ts

```diff
@@ -0,0 +1,113 @@
+import { Plugin } from '@envelop/core';
+import AWSXRay from 'aws-xray-sdk-core';
+import { GraphQLResolveInfo } from 'graphql';
+
+export interface UseXRayOptions<Context> {
+  resolvers: boolean | ((options: { info: GraphQLResolveInfo; context: Context }) => boolean);
+}
+
+const DEFAULT_OPTIONS: UseXRayOptions<unknown> = {
+  resolvers: true,
+};
+
+export enum AttributeName {
+  RESOLVER_FIELD_NAME = 'graphql.resolver.fieldName',
+  RESOLVER_TYPE_NAME = 'graphql.resolver.typeName',
+  RESOLVER_RESULT_TYPE = 'graphql.resolver.resultType',
+}
+
+const tracingSegmentSymbol = Symbol('OPEN_TELEMETRY_GRAPHQL');
+
+export const useXRay = <Context>(rawOptions?: Partial<UseXRayOptions<Context>>): Plugin => {
+  const options: UseXRayOptions<Context> = {
+    ...DEFAULT_OPTIONS,
+    ...(rawOptions || {}),
```

Comment:
```suggestion
    ...rawOptions,
```
An empty object is unnecessary